### ISSUE TITLE
Abstract classes implementing interfaces shouldn't complain about missing methods

### ISSUE THREAD

**Issue by shicks** [google/closure-compiler#2209]
[Example](https://closure-compiler-debugger.appspot.com/#input0%3D%252F**%2520%2540interface%2520*%252F%250Aclass%2520Iface%2520%257B%250A%2520%2520%252F**%2520%2540return%2520%257Bnumber%257D%2520*%252F%2520foo()%2520%257B%257D%250A%257D%250A%250A%252F**%2520%2540abstract%2520%2540implements%2520%257BIface%257D%2520*%252F%250Aclass%2520Abstract%2520%257B%250A%257D%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26CLOSURE_PASS%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1)

```js
/** @interface */
class Interface {
  /** @return {number} */ method() {}
}

/** @abstract @implements {Interface} */
class Abstract {
}
```

produces the error

```
input0:7: WARNING - property method on interface Interface is not implemented by type Abstract
class Abstract {
^^^^^^^^^^^^^^^^^

0 error(s), 1 warning(s), 100.0% typed
```

But since the class is abstract, it should not be a problem.  It should just get an implicit abstract method added instead.  Adding the method explicitly works around the error:

```js
  /** @override @abstract */ method() {}
```

**Comment by gkdn** [comment]
FYI, the problem also exists for OTI.

**Comment by lauraharker** [comment]
This was fixed in https://github.com/google/closure-compiler/commit/393ec71775db7c35e4d3abf4b9eea85d97a54134.

[Debugger link](https://closure-compiler-debugger.appspot.com/#input0%3D%252F**%2520%2540interface%2520*%252F%250Aclass%2520Interface%2520%257B%250A%2520%2520%252F**%2520%2540return%2520%257Bnumber%257D%2520*%252F%2520method()%2520%257B%257D%250A%257D%250A%250A%252F**%2520%2540abstract%2520%2540implements%2520%257BInterface%257D%2520*%252F%250Aclass%2520Abstract%2520%257B%250A%257D%26input1%26conformanceConfig%26externs%26refasterjs-template%26ENABLE_ALL_DIAGNOSTIC_GROUPS%3Dtrue%26CHECK_TYPES%3Dtrue%26CLOSURE_PASS%3Dtrue%26PRESERVE_TYPE_ANNOTATIONS%3Dtrue%26PRETTY_PRINT%3Dtrue)
